### ISSUE TITLE
Property default values missing in output

### ISSUE THREAD

**Issue by voslucas** [jburzynski/TypeGen#55]
TypeGen v2.2.1.
Using GenerationSpec + I don't want to use annotations.

The following class: 

``` csharp
class Test
    {
        public string myFieldString = "Hello";
        public string myPropertyString { get; set; } = "World!";
    }
```
is converted to typescript but it lacks the default value 'World!' for the property myPropertyString. Default property values are available since C# 6.0.

``` typescript
export class Test {
    myFieldString: string = "Hello";
    myPropertyString: string;
}
```

Please advise!

**Comment by jburzynski** [comment]
I didn't know it's possible for properties as well!

I guess it would also make sense to generate the default value even if you specify it in the constructor (and not using the initializer), but I just missed it.

I'll release a patch fixing this and let you know here when it's released.

In the meantime one solution would be to specify values with the `DefaultValue` method in the generation spec, but this would of course only work if you don't have a lot of values to set.

**Comment by jburzynski** [comment]
I just released 2.3.1, which adds support for generating default values for properties.

There is also one change in how default values for properties/fields are generated: right now if a property/field has a default value for its type, this value will not be generated. Quick example below:

```csharp
public class MyClass
{
  public int f1; // default value is not generated in TS
  public int f2 = 0; // default value is not generated in TS, because f2 is assigned default(int)
  public int f3 = 2; // default value is generated in TS
}
```

This is because TypeGen uses reflection and all property/field values are already initialized when you read them.

One solution would be to use Roslyn to check if a given value was assigned by hand, but this would require some major modifications, so it wouldn't be a short-term plan. And also this would only work for sources inside a project, and not for types in external dlls.

--------------------

One minor side effect of generating default TS values for properties is that default values are generated for calculated properties too:

```csharp
public class MyClass
{
  public int P1 { get; set; } = 1;
  public int P2 { get; set; } = 2;
  public int Sum => P1 + P2; // this will be generated to TS with default value of 3
}
```

But I thought this might not be a big problem in most cases, because you typically ignore calculated properties when generating to TS or you don't use them anyway, because their logic is not translated to TS.

--------------------------

More info on how default values are generated in TypeGen is in [the docs](https://typegen.readthedocs.io/en/latest/whatisgenerated.html#default-values-for-typescript-properties).

**Comment by jburzynski** [comment]
closed due to inactivity
